lua/ttt2/libraries/playermodels.lua

```diff
@@ -0,0 +1,190 @@
+		net.SendToServer()
+	end
+end
+
+---
+-- Returns an indexed table with all the models that are in the selction pool.
+-- @return table An indexed table with all selected player models
+-- @realm server
+function playermodels.GetSelectedModels()
+	if SERVER then
+		local playermodelPoolModel = orm.Make(playermodels.sqltable)
+		local playermodelPool = playermodelPoolModel:Where({{column = "state", value = true}}) or {}
+
+		local playerModelPoolNames = {}
+
+		for i = 1, #playermodelPool do
+			if playermodelPool[i].name == playermodels.initializedFlag then continue end
+
+			playerModelPoolNames[#playerModelPoolNames + 1] = playermodelPool[i].name
+		end
+
+		return playerModelPoolNames
+	else
+		return playermodels.selectedModels or {}
+	end
+end
+
+---
+-- Checks if a provided model is in the selection pool.
+-- @note While this function is shared, the data is only available for superadmin on the
+-- client, if no manual sync is triggered.
+-- @param string name The name of the model
+-- @return boolean Returns true, if the model is in the selection pool
+-- @realm shared
+function playermodels.HasSelectedModel(name)
+	local models = playermodels.GetSelectedModels()
+
+	for i = 1, #models do
+		if models[i] ~= name then continue end
+
+		return true
+	end
+
+	return false
+end
+
+
+if CLIENT then
+	net.ReceiveStream("TTT2StreamModelTable", function(data)
+		playermodels.selectedModels = data
+	end)
+end
+
+if SERVER then
+	---
+	-- @realm server
+	local cvCustomModels = CreateConVar("ttt2_use_custom_models", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+
+	local initialModels = {
+		["css_phoenix"] = true,
+		["css_arctic"] = true,
+		["css_guerilla"] = true,
+		["css_leet"] = true
+	}
+
+	-- extend the playermodels scope on the server
+	playermodels.sqltable = "ttt2_playermodel_pool"
+	playermodels.savingKeys = {
+		state = {typ = "bool", default = false}
+	}
+	playermodels.initializedFlag = "__INITIALIZED__"
+
+	playermodels.fallbackModel = "models/player/phoenix.mdl"
+
+	---
+	-- Initializes the database. This adds all models and sets the default models to true, if it
+	-- is the first init on this server.
+	-- @return boolean Returns false, if it failed during the process
+	-- @internal
+	-- @realm server
+	function playermodels.InitializeDatabase()
+		if not sql.CreateSqlTable(playermodels.sqltable, playermodels.savingKeys) then
+			return false
+		end
+
+		local playermodelPoolModel = orm.Make(playermodels.sqltable)
+		local initializedFlag = playermodelPoolModel:Find(playermodels.initializedFlag)
+		local isNewTable = initializedFlag and (initializedFlag.state ~= true) or true
+
+		for name in pairs(playerManagerAllValidModels()) do
+			if playermodelPoolModel:Find(name) then continue end
+
+			playermodelPoolModel:New({name = name, state = (isNewTable and initialModels[name]) or false}):Save()
+		end
+
+		-- mark table as initialized
+		playermodelPoolModel:New({name = playermodels.initializedFlag, state = true}):Save()
```

Comment:
This really is pretty hacky and is not even needed in my opinion. This code would do the same without it, wouldn't it?

Reply:
no, because `New` overwrites existing entries. Therefore I have to check if the table is initialized

Reply:
And that's no problem since you are already checking with find if the exist.

Reply:
I see, yes. I think this can be simplified